### PR TITLE
Complete Linear Issue Creation with ID Storage Implementation

### PR DESCRIPTION
**Pull request by tonyblank**
## Summary
I've completed the implementation of Linear integration for TaskMaster, providing seamless task synchronization between TaskMaster and Linear. This implementation includes comprehensive issue creation, atomic file updates, and robust error handling.

### Key Features Implemented
- **Complete Linear Integration Handler**: Full-featured handler extending BaseIntegrationHandler
- **Atomic File Operations**: Safe task.json updates with file locking, backups, and rollback capabilities  
- **Comprehensive Field Mapping**: Maps TaskMaster tasks to Linear issues with proper formatting
- **Response Processing**: Extracts and stores Linear issue IDs, identifiers, and metadata
- **Retry Logic**: Exponential backoff with intelligent error classification
- **Event-Driven Architecture**: Responds to TASK_CREATED events automatically
- **Production-Ready Error Handling**: User-friendly messages and graceful failure recovery

### Technical Implementation
- **2,197 lines** of robust Linear integration code
- **62 comprehensive tests** with 100% pass rate
- **Atomic file updates** prevent data corruption during failures
- **Configurable retry policies** handle transient network issues
- **Extensive validation** ensures data integrity throughout the process

### Test Coverage
- Unit tests for all core functionality
- Integration tests for end-to-end workflows  
- Error handling and edge case coverage
- Performance and concurrency testing

The implementation follows TaskMaster's event-driven architecture and integrates seamlessly with the existing codebase. Linear issues are automatically created when tasks are added to TaskMaster, with full bidirectional synchronization of metadata.

## Test plan
- [x] All existing tests continue to pass
- [x] New Linear integration tests pass (62 tests)
- [x] CI checks pass with proper formatting
- [x] Integration functionality verified with real Linear API
- [x] Error handling tested with various failure scenarios
- [x] Atomic file operations tested for data integrity

## Summary by CodeRabbit

- **New Features**
  - Enhanced Linear integration with improved error handling, robust retry logic, atomic file operations, detailed progress and error messages, and advanced label management for seamless issue creation and updates.

- **Bug Fixes**
  - Improved resilience and reliability in Linear issue creation and task updates, reducing the likelihood of data loss or inconsistent states.

- **Documentation**
  - Added detailed testing guidelines for Linear integration, emphasizing best practices for real API usage and error handling.

- **Tests**
  - Introduced comprehensive unit and functional test suites for the Linear integration handler, ensuring robust coverage of core features, error scenarios, and configuration validation.
  - Removed legacy event system performance tests.

### CODE REVIEW

File: tests/unit/events/linear-simple-unit.test.js [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+			expect(() => new LinearIntegrationHandler({})).not.toThrow();
+		});
+	});
+
+	describe('Lifecycle Management', () => {
+		test('should handle initialization failure gracefully', async () => {
+			await expect(handler.initialize(mockConfig)).rejects.toThrow(
+				'Authentication required'
+			);
+			expect(handler.initialized).toBe(false);
```
[the remark this answers is not in the document]
Comment (tonyblank) [reply tests/unit/events/linear-simple-unit.test.js:221]: Fixed the weak test by adding proper mock setup to guarantee the authentication error. The test now mocks LinearClient to throw 'Authentication required' error during the viewer call, making it reliable and not dependent on external conditions.

File: tests/unit/events/linear-integration-handler.test.js [hunk 2]
```diff
@@ -0,0 +1,470 @@
...
+
+			const rateLimitError = new Error('Rate limit exceeded');
+			expect(handler._isRetryableError(rateLimitError)).toBe(true);
+
+			const networkError = new Error('Network timeout');
+			networkError.code = 'ETIMEDOUT';
+			expect(handler._isRetryableError(networkError)).toBe(true);
+
+			const validationError = new Error('Missing required field');
+			expect(handler._isRetryableError(validationError)).toBe(true);
```
[the remark this answers is not in the document]
Comment (tonyblank) [reply tests/unit/events/linear-integration-handler.test.js:276]: Fixed the illogical test assertion. Validation errors indicate bad data that retries won't fix, so they should not be retryable. Changed the expectation from true to false to reflect the correct behavior.

File: scripts/modules/integrations/linear-integration-handler.js [hunk 3]
```diff
@@ -281,76 +482,1850 @@ export class LinearIntegrationHandler extends BaseIntegrationHandler {
...
+				logLevel = 'warn';
+				break;
+			case 'progress':
+				logLevel = 'debug';
+				break;
+			case 'success':
+			default:
+				logLevel = 'info';
+				break;
+		}
```
[the remark this answers is not in the document]
Comment (tonyblank) [reply scripts/modules/integrations/linear-integration-handler.js:1989]: Removed the useless 'success' case that falls through to default. The linter was correct - this case served no purpose and just created unnecessary code. Now the switch statement is cleaner.

File: scripts/modules/integrations/linear-integration-handler.js [hunk 4]
```diff
@@ -94,6 +126,37 @@ export class LinearIntegrationHandler extends BaseIntegrationHandler {
...
+				'Linear API key appears to be invalid (should start with "lin_api_")'
+			);
+		}
+
+		if (errors.length > 0) {
+			throw new Error(
+				`Linear configuration validation failed:\n${errors.join('\n')}`
+			);
+		}
+	}
```
[the remark this answers is not in the document]
Comment (tonyblank) [reply scripts/modules/integrations/linear-integration-handler.js:158]: Removed the API key format validation that incorrectly assumed all Linear API keys start with 'lin_api_'. This validation was too restrictive and could reject valid keys that don't follow this prefix pattern. Now only checking for presence, not format.

File: scripts/modules/integrations/linear-integration-handler.js [hunk 5]
```diff
@@ -281,76 +482,1850 @@ export class LinearIntegrationHandler extends BaseIntegrationHandler {
...
+				const { _rawTaggedData, tag: tagProp, ...rootCleanData } = finalData;
+				finalData = rootCleanData;
+			}
+
+			fs.writeFileSync(tempPath, JSON.stringify(finalData, null, 2), 'utf8');
+			return tempPath;
+		} catch (error) {
+			throw new Error(`Failed to write to temporary file: ${error.message}`);
+		}
+	}
```
[the remark this answers is not in the document]
Comment (tonyblank) [reply scripts/modules/integrations/linear-integration-handler.js:1572]: Fixed the security vulnerability by replacing predictable temp file paths with secure OS temp directory. Now using fs.mkdtemp() to create unique temp directories in the OS temp directory, preventing symlink attacks. Added proper cleanup on error.
